tests/components/bond/test_init.py

```diff
@@ -18,6 +23,17 @@ async def test_async_setup_no_domain_config(hass: HomeAssistant):
     assert result is True
 
 
+async def test_async_setup_raises_entry_not_ready(hass: HomeAssistant):
+    """Test that it throws ConfigEntryNotReady when exception occurs during setup."""
+    config_entry = MockConfigEntry(
+        domain=DOMAIN, data={CONF_HOST: "some host", CONF_ACCESS_TOKEN: "test-token"},
+    )
+
+    with patch_bond_version(side_effect=ClientConnectionError()):
+        with pytest.raises(ConfigEntryNotReady):
+            await async_setup_entry(hass, config_entry)
```

Comment:
Please start all tests with the core interfaces. Here use `hass.config_entries.async_setup`. Then assert the config entry state to be `ENTRY_STATE_SETUP_RETRY`

Reply:
Sure, PR is here: https://github.com/home-assistant/core/pull/38303